### PR TITLE
Unify build-script running to always use rpmfcExec()

### PR DESCRIPTION
**Pull request by ffesti**
Extends #659

### CODE REVIEW

File: build/rpmfc.c [hunk 1]
```diff
@@ -248,23 +248,24 @@ static rpmds rpmdsSingleNS(rpmstrPool pool,
...
  * @param buildRoot	buildRoot directory (or NULL)
+ * @param dup		duplicate output to stream (or NULL)
  * @return		buffered stdout from script, NULL on error
  */     
-static StringBuf getOutputFrom(ARGV_t argv,
-                        const char * writePtr, size_t writeBytesLeft,
-                        int failNonZero, const char *buildRoot)
+static int getOutputFrom(ARGV_t argv,
+			 const char * writePtr, size_t writeBytesLeft,
+			 StringBuf sb_stdout,
```
Comment (ignatenkobrain) [inline build/rpmfc.c]: you've missed `sb_stdout` in documentation for this function.

File: build/rpmfc.c [hunk 2]
```diff
@@ -248,23 +248,24 @@ static rpmds rpmdsSingleNS(rpmstrPool pool,
  * @param writeBytesLeft no. of bytes to feed to script on stdin
  * @param failNonZero	is script failure an error?
  * @param buildRoot	buildRoot directory (or NULL)
+ * @param dup		duplicate output to stream (or NULL)
  * @return		buffered stdout from script, NULL on error
```
Comment (ignatenkobrain) [inline build/rpmfc.c]: This one should be updated too, since it is now return code.

File: build/rpmfc.c [hunk 3]
```diff
@@ -248,23 +248,24 @@ static rpmds rpmdsSingleNS(rpmstrPool pool,
...
  */     
-static StringBuf getOutputFrom(ARGV_t argv,
-                        const char * writePtr, size_t writeBytesLeft,
-                        int failNonZero, const char *buildRoot)
+static int getOutputFrom(ARGV_t argv,
```
Comment (ignatenkobrain) [inline build/rpmfc.c]: probably I would make this function to return rpmRC rather than (int), but up to you.

Comment (pmatilai) [reply build/rpmfc.c]: Actually rpmRC is wrong for 90% of the uses it's used for, especially in the build code. It's only really meaningful for package/header read operations, for pretty much everything else int is better.

**Comment by ffesti** [comment]
We have doc strings for internal helper functions? This is madness...

OK, fixed them...

**Pull request merged**